keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/annealing/ResourceActuator.kt

```diff
@@ -0,0 +1,84 @@
+package com.netflix.spinnaker.keel.annealing
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.persistence.ResourceRepository
+import com.netflix.spinnaker.keel.plugin.ResourceConflict
+import com.netflix.spinnaker.keel.plugin.ResourceHandler
+import com.netflix.spinnaker.keel.plugin.supporting
+import de.danielbechler.diff.ObjectDifferBuilder
+import de.danielbechler.diff.node.DiffNode
+import org.slf4j.LoggerFactory
+import org.springframework.context.event.EventListener
+import org.springframework.scheduling.annotation.Async
+import org.springframework.stereotype.Component
+
+@Component
+class ResourceActuator(
+  private val resourceRepository: ResourceRepository,
+  private val handlers: List<ResourceHandler<*>>
+) {
+
+  private val differ = ObjectDifferBuilder.buildDefault()
+
+  @Async
```

Comment:
This is pretty neat because it means that even though the event dispatch is synchronous, this handler runs async. In other words it won't block that loop in `ResourceActuationScheduler`. This work will get run on a thread pool.